In [1]:
import numpy as np
import pandas as pd
from io import StringIO
import warnings
warnings.filterwarnings('ignore')
import acquire
import prepare
# grab Decision Tree Classifier and some helping friends from sklearn.tree
from sklearn.tree import DecisionTreeClassifier, plot_tree, export_text
# get classification report for model assessment
from sklearn.metrics import classification_report

Using the titanic data, in your classification-exercises repository, create a notebook, model.ipynb where you will do the following:


1. What is your baseline prediction? What is your baseline accuracy? remember: your baseline prediction for a classification problem is predicting the most prevelant class in the training dataset (the mode). When you make those predictions, what is your accuracy? This is your baseline accuracy.

In [7]:
#acquire train data 
df = acquire.get_titanic_data()
train, validate, test = prepare.prep_titanic_data(df)
train.shape, validate.shape, test.shape

((534, 10), (178, 10), (179, 10))

In [8]:
train.head()

passenger_id  survived  pclass  sibsp  parch      fare  alone  sex_male  \
455           455         1       3      0      0    7.8958      1         1   
380           380         1       1      0      0  227.5250      1         0   
492           492         0       1      0      0   30.5000      1         1   
55             55         1       1      0      0   35.5000      1         1   
243           243         0       3      0      0    7.1250      1         1   

     embark_town_Queenstown  embark_town_Southampton  
455                       0                        0  
380                       0                        0  
492                       0                        1  
55                        0                        1  
243                       0                        1

In [10]:
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 534 entries, 455 to 496
Data columns (total 10 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   passenger_id             534 non-null    int64  
 1   survived                 534 non-null    int64  
 2   pclass                   534 non-null    int64  
 3   sibsp                    534 non-null    int64  
 4   parch                    534 non-null    int64  
 5   fare                     534 non-null    float64
 6   alone                    534 non-null    int64  
 7   sex_male                 534 non-null    uint8  
 8   embark_town_Queenstown   534 non-null    uint8  
 9   embark_town_Southampton  534 non-null    uint8  
dtypes: float64(1), int64(6), uint8(3)
memory usage: 34.9 KB


In [12]:
# X is going to be our entire domain, indepedent variables, minus the target variable
# y is going to be our target variable, the dependent thing that we are using X to predict
X_train = train.drop(columns = "survived")
y_train = train.survived
X_validate = validate.drop(columns = "survived")
y_validate = validate.survived
X_test = test.drop(columns = "survived")
y_test = test.survived

In [13]:
#check X_train: independent variable
X_train.head()

passenger_id  pclass  sibsp  parch      fare  alone  sex_male  \
455           455       3      0      0    7.8958      1         1   
380           380       1      0      0  227.5250      1         0   
492           492       1      0      0   30.5000      1         1   
55             55       1      0      0   35.5000      1         1   
243           243       3      0      0    7.1250      1         1   

     embark_town_Queenstown  embark_town_Southampton  
455                       0                        0  
380                       0                        0  
492                       0                        1  
55                        0                        1  
243                       0                        1

In [14]:
#check the target variable
y_train.head()

455    1
380    1
492    0
55     1
243    0
Name: survived, dtype: int64

In [15]:
#value counts the target variable
y_train.value_counts()

0    329
1    205
Name: survived, dtype: int64

In [16]:
y_train.value_counts(normalize = True)

0    0.616105
1    0.383895
Name: survived, dtype: float64

baseline model is the mode of target varible, which is did not survive in this case.
baseline accuracy is the percent value of baseline model here which is 61.61% here

2. Fit the decision tree classifier to your training sample and transform (i.e. make predictions on the training sample)

In [26]:
# create decision tree
clf = DecisionTreeClassifier(max_depth=3, random_state=123)

In [27]:
#fit the random forest algorithm on to training sample
clf.fit(X_train,y_train)

DecisionTreeClassifier(max_depth=3, random_state=123)

In [32]:
#make prediction on train observation
y_pred = clf.predict(X_train)
y_pred

array([0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1,
       0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1,
       0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 1,
       1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0,
       0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1,
       1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1, 0,
       1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0,

In [37]:
#estimate probability
y_pred_proba = clf.predict_proba(X_train)
y_pred_proba

array([[0.90322581, 0.09677419],
       [0.        , 1.        ],
       [0.61458333, 0.38541667],
       ...,
       [0.90322581, 0.09677419],
       [0.90322581, 0.09677419],
       [0.        , 1.        ]])

#3. Evaluate your in-sample results using the model score, confusion matrix, and classification report.